In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
from matplotlib import pyplot as plt
%matplotlib inline

In [2]:
df = pd.read_csv('score_prediction.csv')
df.head()

Hours  Scores
0    2.5      21
1    5.1      47
2    3.2      27
3    8.5      75
4    3.5      30

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Hours   25 non-null     float64
 1   Scores  25 non-null     int64  
dtypes: float64(1), int64(1)
memory usage: 528.0 bytes


In [4]:
df.isnull().sum()

Hours     0
Scores    0
dtype: int64

# WHAT WILL BE THE PREDICTED SCORE , IF THE STUDENT STUDIES FOR 9.25 HOURS PER DAY?

In [5]:
df[df['Hours']==9.25]

Empty DataFrame
Columns: [Hours, Scores]
Index: []

In [6]:
X = np.array(df[['Hours']])
y = df['Scores']

In [7]:
X

array([[2.5],
       [5.1],
       [3.2],
       [8.5],
       [3.5],
       [1.5],
       [9.2],
       [5.5],
       [8.3],
       [2.7],
       [7.7],
       [5.9],
       [4.5],
       [3.3],
       [1.1],
       [8.9],
       [2.5],
       [1.9],
       [6.1],
       [7.4],
       [2.7],
       [4.8],
       [3.8],
       [6.9],
       [7.8]])

In [8]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=1)

In [9]:
from sklearn.linear_model import LinearRegression
lr= LinearRegression()

In [10]:
lr.fit(x_train , y_train)

LinearRegression()

In [11]:
y_pred = lr.predict(x_test)

In [15]:
a = pd.DataFrame({'actual_value': y_pred , 'predicted_value': y_test})
a

actual_value  predicted_value
14      9.970262               17
13     32.984700               42
17     18.339148               24
3      87.382463               75
21     48.676362               54

In [18]:
# NOW THE GIVEN VALUE IS 9.25 HOURS
given_hours = np.array([[9.25]])

In [20]:
prediction = lr.predict(given_hours)
prediction

array([95.22829438])

# CONCLUSION : 
#SCORE PREDICTED WHEN A STUDENT STUDIES FOR 9.25 HOURS IS 95.2